File: core/Application.cs
namespace core{
    /// <summary>
    /// Class that contains all application information (e.g. settings)
    /// </summary>
    public sealed class Application{
        /// <summary>
        /// ApplicationSettings with the current application settings
        /// </summary>
        private static readonly ApplicationSettings applicationSettings=new ApplicationSettings();

        /// <summary>
        /// Returns the current application settings
        /// </summary>
        /// <returns>ApplicationSettings with the current application settings</returns>
        public static ApplicationSettings settings(){return applicationSettings;}

        /// <summary>
        /// Hides default constructor
        /// </summary>
        private Application(){}
    }
}

File: MYCA/myca/core/src/main/java/cdiomyc/core/application/Application.java
package cdiomyc.core.application;

/**
 * Class that holds all global application context settings & dependencies
 * @author <a href="https://github.com/freitzzz">freitzzz</a>
 */
public final class Application {
    /**
     * ApplicationSettings with the current application settings
     */
    private static final ApplicationSettings APPLICATION_SETTINGS=new ApplicationSettings();
    /**
     * Returns the current application settings
     * @return ApplicationSettings with the current application settings
     */
    public static ApplicationSettings settings(){return APPLICATION_SETTINGS;}
}
